cmd/router/main.go

```diff
@@ -38,8 +38,13 @@ import (
 
 var log = logf.Log.WithName("InferenceGraphRouter")
 
-func callService(serviceUrl string, input []byte) ([]byte, error) {
-	resp, err := http.Post(serviceUrl, "application/json", bytes.NewBuffer(input))
+func callService(serviceUrl string, input []byte, headers http.Header) ([]byte, error) {
+	client := http.Client{}
+	req, err := http.NewRequest("POST", serviceUrl, bytes.NewBuffer(input))
+	req.Header = headers // propagating the headers to steps i.e. to ISVCs
```

Comment:
Does it always make sense to propagate all the headers to the services or we need to filter for certain headers.

Reply:
I don't see a need to why would we want to filters the headers to be propagated. 
Do you have a use case in mind ?

Reply:
Like the span Id used for distributed tracing which should not be passed to the next step. I guess we need to explicitly support distributed tracing in graph orchestrator so it can propagate headers correctly.

Reply:
Ok. Since router (or better to say IG) doesn't support distributed tracing at this time point in time, so we have few approaches that we can proceed with:

1. Since router (or IG) doesn’t support distributed tracing at this time, so we go ahead with propagating all the headers and then the work for enabling distributed tracing (which will be a separate PR) should take care of remove tracing related headers and keep the rest as it is. Basically keeping things backward compatible.
Passing all the headers at this point in time anyways won't break distributed tracing since router is not instrumented for distributed tracing anyways. Is this understanding Right ? 

2. Second approach is, we do pass on only selective headers. I checked how ApolloGraphQL does it for it's router (see [here](https://www.apollographql.com/docs/router/configuration/header-propagation/))
This work might be scrapped later on once router code is instrumented for distributed tracing.

Let me know your thoughts on both the approaches.

Reply:
We actually run into some issue with propagating headers between transformer and predictor in the other related PR as it accidentally copied over the `HOST` header from the previous request which messes up the calls, I think it is still better to explicitly copy over the headers which are the same across the request chain instead of copying everything which can be very hard to debug if some header is passed along unexpectedly.

Reply:
@yuzisun : I have made passing headers configurable using `router` section of `inferenceservice-config` config map.
This is how headers to propagated can be mentioned in router's config.
```
{
    "image" : "docker.io/rachitchauhan885/router-20acce65bc84b6e5b28ae59ce41635d3@sha256:729d6c170f28355075dd609dd2e101e8ce519342e406982b906364107703660e",
    "memoryRequest": "100Mi",
    "memoryLimit": "500Mi",
    "cpuRequest": "100m",
    "cpuLimit": "100m",
    "headers": {
      "propagate":[
        "Authorization"
      ]
    }
}
```